spirv_glsl.cpp

```diff
@@ -9189,6 +9212,18 @@ string CompilerGLSL::access_chain_internal(uint32_t base, const uint32_t *indice
 					break;
 				}
 			}
+			else if (backend.force_meged_mesh_block_hlsl && i == 0 && var && !is_builtin_variable(*var) &&
```

Comment:
Is the motivation for this branch that all stage IO variables must go through MeshPrimitives/MeshVertices in HLSL?

Reply:
As breathy discussed in the ticket:
HLSL supports only one `vertices`/`primitives` block, so all user-defined stuff has to be merged with system-defined outputs.
Another thing that I forgot to mention: I don't know for sure why spirv-cross flattens output blocks. Flattening creates very awkward cases for mesh-shader.

Reply:
Mostly because some backends (e.g. MSL) don't support nested structs in I/O blocks. Note that MSL *doesn't* flatten I/O blocks for tessellation shaders; it should be possible to do something similar for mesh shaders (both MSL and HLSL).